examples/vision/edsr.py

```diff
@@ -0,0 +1,320 @@
+"""
+Title: Enhanced Deep Residual Networks for Single Image Super-Resolution
+Author: Gitesh Chawda
+Date created: 16-03-2022
+Last modified: 16-03-2022
+Description: Implementing EDSR model on DIV2K Dataset.
+"""
+
+"""
+## Introduction 
+
+Enhanced Deep Residual Networks for Single Image Super-Resolution
+[EDSR](https://arxiv.org/abs/1707.02921) by Bee Lim, Sanghyun Son, Heewon Kim, Seungjun
+Nah, Kyoung Mu Lee.
+
+The EDSR architecture is based on the SRResNet architecture, consisting of multiple
+residual blocks. It doesn't use batch normalization layers (input and output have similar
+distributions, normalizing intermediate features may not be desirable) instead it uses
+constant scaling layers to ensure stable training. An L1 loss function (absolute error)
+is used instead of L2 (MSE), the authors showed better performance empirically and it
+requires less computation.
+
+In this code example we will implementing base model that includes just 16 ResBlocks and
+64 channels.
+
+Alernatively, you can also build a ESPCN Model as shown in the Keras example [Image
+Super-Resolution using an Efficient Sub-Pixel CNN
+](https://keras.io/examples/vision/super_resolution_sub_pixel/#image-superresolution-using
```

Comment:
Make sure to fit markdown links on a single line. See guide: https://github.com/keras-team/keras-io/blob/master/contributor_guide.md